features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/gson/GsonExt.kt

```diff
@@ -6,37 +6,56 @@
 
 package com.datadog.android.sessionreplay.internal.gson
 
+import com.datadog.android.api.InternalLogger
 import com.google.gson.JsonArray
 import com.google.gson.JsonElement
 import com.google.gson.JsonObject
 import com.google.gson.JsonPrimitive
 
+internal const val BROKEN_JSON_ERROR_MESSAGE =
+    "SR GsonExt: Unable parse the batch data into a JsonObject"
+
 @Suppress("SwallowedException")
-internal fun JsonElement.safeGetAsJsonObject(): JsonObject? {
+internal fun JsonElement.safeGetAsJsonObject(internalLogger: InternalLogger): JsonObject? {
     return try {
         asJsonObject
     } catch (e: IllegalStateException) {
-        // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
+        internalLogger.log(
+            InternalLogger.Level.ERROR,
+            InternalLogger.Target.TELEMETRY,
+            { BROKEN_JSON_ERROR_MESSAGE },
```

Comment:
Since you want to send this to telemetry, you should consider adding a more explicit message, as we're gonna parse lot of messages : 
e.g.
```
internal const val BROKEN_JSON_ERROR_MESSAGE =
    "SR GsonExt: Unable parse the batch data into a JsonObject: expected to parse %s as %s "
```

and then 

```suggestion
            { BROKEN_JSON_ERROR_MESSAGE.format(Locale.US, this.toString(), "JsonObject" },
```

Reply:
I was thinking about that but then I thought I am going to create a very high cardinality with this. What do you think ?